lib/post_revisor.rb

```diff
@@ -113,7 +113,7 @@ def self.track_and_revise(topic_changes, field, attribute)
         DB.after_commit do
           post = tc.topic.ordered_posts.first
           notified_user_ids = [post.user_id, post.last_editor_id].uniq
-          Jobs.enqueue(:notify_tag_change, post_id: post.id, notified_user_ids: notified_user_ids, diff_tags: ((tags - prev_tags) | (prev_tags - tags)))
+          Jobs.enqueue(:notify_tag_change, post_id: post.id, notified_user_ids: notified_user_ids, diff_tags: ((tags - prev_tags) | (prev_tags - tags))) if !SiteSetting.disable_tags_edit_notifications
```

Comment:
1/ That line is getting a bit too long. I think this is one of the case where it's worth being "explicit" and rewrite this like so

```ruby
if !SiteSetting.disable_tags_edit_notifications
  Jobs.enqueue(:notifify_tag_change,
    post_id: post.id,
    ...
   )
end

2/ What about jobs that are currently in the queue when the site setting is changed? We should probably add an early return in the job itself as well

Reply:
That makes sense. I assumed that it does not matter too much if the site setting changes because the job usually runs fast anyway. However, this logic can help in some cases, especially when the server is overloaded. I submitted a new PR to fix that.